### PR TITLE
[ML] Put ML filter API response should contain the filter

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutFilterAction.java [hunk 1]
```diff
@@ -115,10 +115,53 @@ public RequestBuilder(ElasticsearchClient client) {
...
+    public static class Response extends ActionResponse implements ToXContentObject {
 
-        public Response() {
-            super(true);
+        private MlFilter filter;
+
+        Response() {
+        }
+
+        public Response(MlFilter calendar) {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutFilterAction.java]: Rename `calendar` -> `filter`

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/PutFilterActionResponseTests.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.action;
+
+import org.elasticsearch.test.AbstractStreamableTestCase;
+import org.elasticsearch.xpack.core.ml.job.config.MlFilterTests;
+
+public class PutFilterActionResponseTests extends AbstractStreamableTestCase<PutFilterAction.Response> {
```
Comment (davidkyle) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/PutFilterActionResponseTests.java]: Can this be a `AbstractStreamableXContentTestCase`? The toXContent method calls `filter.toXContent()` so you can use the filter parser

**Review by davidkyle — APPROVED** [review]
LGTM. Left a nit and a suggestion

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutFilterAction.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/PutFilterAction.java]: oups :-)

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/PutFilterActionResponseTests.java (hunk 2 above)
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/action/PutFilterActionResponseTests.java]: Sure. I also did it for the `PutCalendarActionResponseTests`.

**Pull request merged**